src/main/java/roboy/util/TelegramCommunicationHandler.java

```diff
@@ -0,0 +1,339 @@
+                        Log.error(this, "Message receiving has been interrupted.");
+                        e.printStackTrace();
+                    }
+                }
+            }
+        }else{
+            //The update does not have messages
+            Log.error(this,"update: "+update.toString());
+        }
+    }
+
+    /**
+     * Called when specified time passed w.r.t. unique chatID.
+     * Concatenate all the messages that is not processed.
+     * Calls the InputDevice for telegram
+     * @param chatID unique identifier for a chat.
+     */
+    @Override
+    public void onTimeout(String chatID) {
+
+        // timeout method for getting all the messages
+        //telegramTimeout.stop();
+        for(Timeout t: telegramTimeouts){
+            if(t.getUnique().equals(chatID)){
+                t.stop();
+            }
+        }
+
+        List<HashMap<String, String>> removedObjects = new ArrayList<>();
+
+        // get the all messages
+        HashMap<String, String> result = null;
+        for(HashMap<String, String> p: pairs){
+            //Map a = new HashMap<String, String>();
+            String chatIdFromPair = p.keySet().toArray()[0].toString();
+            if(!chatID.equals(chatIdFromPair)){
+                continue;
+            }
+
+            removedObjects.add(p);
+            String message = p.values().toArray()[0].toString();
+
+            // check if the result initialized
+            if(result == null) {
+                result = new HashMap<>();
+                result.put(chatID, message);
+            }
+            else {
+                // sum all of the messages
+                String oldMessage = result.values().toArray()[0].toString();
+                String newMessage = oldMessage + " " + message;
+
+                //equal chat id
+                result = new HashMap<>();
+                result.put(chatID, newMessage);
+            }
+        }
+
+
+        if(result != null) {
+            // notify the input device
+            pairs.removeAll(removedObjects);
+            TelegramInput.onUpdate(result);
+        }
+    }
+
+    // ---- SEND METHODS ----
+
+    /**
+     * Called from the OutputDevice when a message desired to send
+     * Initiates the "typing status" and waits for a specified time
+     * Sends the message afterwards
+     * @param chatID unique identifier for a chat.
+     */
+    public void sendMessage(String message, String chatID){
+        try {
+            sendTypingFromChatID(chatID);
+
+            TimeUnit.SECONDS.sleep(TYPING_TIME_LIMIT);
+
+            SendMessage sendMessageRequest = new SendMessage();
+            sendMessageRequest.setChatId(chatID);//who should get the message? the sender from which we got the message...
+            sendMessageRequest.setText(message);
+            try {
+                execute(sendMessageRequest);
+            } catch (TelegramApiException e) {
+                //do some error handling
```

Comment:
this shouldn't be empty
add a log output at least

Reply:
@polinominom